tests/integration/test_extract.py

```diff
@@ -15,99 +15,102 @@
+    assert np.array_equal(res_np, res_num)
 
-@pytest.mark.parametrize("arr", ARR, ids=str)
-def test_extract_bool(arr):
-    arr_np = np.array(arr)
-    condition_np = arr_np != 0
-    check_extract(condition_np, arr_np)
-    check_extract(condition_np.flatten(), arr_np)
-    check_extract(condition_np, arr_np.flatten())
-    check_extract(condition_np.swapaxes(0, condition_np.ndim - 1), arr_np)
-    check_extract(condition_np, arr_np.swapaxes(0, condition_np.ndim - 1))
 
+@pytest.mark.xfail
+def test_empty_array():
+    res_np = np.extract([0], [])  # return []
+    res_num = num.extract(
+        [0], []
+    )  # ValueError: arr array and condition array must be of same size
+    assert np.array_equal(res_np, res_num)
 
-@pytest.mark.parametrize("arr", ARR, ids=str)
-def test_extract_nonzero(arr):
-    arr_np = np.array(arr)
-    condition_np = arr_np.copy()
-    check_extract(condition_np, arr_np)
-    check_extract(condition_np.flatten(), arr_np)
-    check_extract(condition_np, arr_np.flatten())
-    check_extract(condition_np.swapaxes(0, condition_np.ndim - 1), arr_np)
-    check_extract(condition_np, arr_np.swapaxes(0, condition_np.ndim - 1))
 
+@pytest.mark.xfail
+def test_none_condition():
+    a = num.array([1, 2, 3, 4])
+    res_np = np.extract(None, a)  # all return []
+    res_num = num.extract(
+        None, a
+    )  # AttributeError: 'NoneType' object has no attribute 'size'
+    assert np.array_equal(res_np, res_num)
 
-VALUES = [
-    [11, 12, 13],
-    [99, 93, 76, 65, 76, 87, 43, 23, 12, 54, 756, 2345, 232, 2323, 12145],
-    [42],
-    [True, False, False, True],
-    [42.3, 42.3, 42.3, 42.3, 42.3, 42.3, 42.3, 42.3],
-    [42 + 3j],
-]
 
+@pytest.mark.parametrize(
+    "con", (-3, 0, 3, False, True, [2], [2, 3], [2, -3, 4], [1, 2, 3, 4, 5])
+)
+def test_negative_condition(con):
+    a = num.array([1, 2, 3, 4])
+    with pytest.raises(ValueError):
+        num.extract(con, a)
+
+
+@pytest.mark.parametrize("shape", SIZES, ids=str)
+def test_extract_basic(shape):
+    np_arr = mk_seq_array(np, shape)
+    num_arr = mk_seq_array(num, shape)
+    # make sure condition is between 1 and 2
+    np_condition = mk_seq_array(np, shape) % 2
+    num_condition = mk_seq_array(num, shape) % 2
+
+    res_np = np.extract(np_condition, np_arr)
+    res_num = num.extract(num_condition, num_arr)
+    assert np.array_equal(res_num, res_np)
 
-@pytest.mark.parametrize("arr", ARR, ids=str)
+
+@pytest.mark.parametrize("shape", SIZES, ids=str)
 @pytest.mark.parametrize("vals", VALUES, ids=str)
-def test_place(arr, vals):
-    arr_np = np.array(arr)
-    vals_np = np.array(vals).astype(arr_np.dtype)
-    condition_np = arr_np != 0
+def test_place_basic(shape, vals):
+    arr_np = mk_seq_array(np, shape)
+    arr_num = num.array(mk_seq_array(num, shape))
 
-    arr_num = num.array(arr_np)
-    condition_num = num.array(condition_np)
+    condition_np = mk_seq_array(np, shape) % 3
+    condition_num = mk_seq_array(num, shape) % 3
```

Comment:
the second parameter or `np.place` is:
```
mask : array_like
    Boolean mask array. Must have the same size as `a`.
```
so it would be good to rename to `mask_np` etc It is a bit confusing when then parameter names do not correspond. But also, why `%3` ? That will give values beween 0, 1, 2 but the docstring says a "Boolean mask array" Are you relying on an automatic type conversion to bool?

Reply:
Fix them.